### PR TITLE
Modify project Infra

### PR DESCRIPTION
**Pull request by fschuch**
## Summary by CodeRabbit

- **New Features**
	- Introduced workflows for release preparation, tagging releases, link checks, continuous integration, and documentation automation.
	- Added guidelines on how to contribute to the project.
	- Updated Visual Studio Code recommendations for Python development and spell checking.
	- Enhanced project documentation and examples, including a new "What’s New" section.

- **Bug Fixes**
	- Corrected a typo in integration tests and fixed a TypeError in the sandbox module.
	- Addressed various bugs and improvements in functions and classes, enhancing functionalities and support for boundary conditions, mesh handling, and more.

- **Documentation**
	- Restructured README for better clarity and updated installation instructions.
	- Improved documentation structure, consistency, and added a references section.

- **Refactor**
	- Major codebase refactor for readability, consistency, and better error handling across various modules and tests.
	- Simplified version computation and reorganized imports and module references.

- **Style**
	- Adopted black code style for a more consistent codebase.

- **Tests**
	- Expanded unit and integration tests, including the transition from `unittest` to `pytest` for parameterized testing and improved test setups.

- **Chores**
	- Updated `.gitignore` and `.pre-commit-config.yaml` with new configurations for project maintenance.

### CODE REVIEW

File: xcompact3d_toolbox/io.py [hunk 1]
```diff
@@ -232,41 +218,49 @@ def get_name_from_filename(self, filename: str) -> str:
...
+        stack_scalar : bool
+            When :obj:`True`, the scalar fields will be stacked in a new coordinate ``n``, otherwise returns one
+            array per scalar fraction (default is :obj:`False`).
+        stack_velocity : bool
+            When :obj:`True`, the velocity will be stacked in a new coordinate ``i`` , otherwise returns one
+            array per velocity component (default is :obj:`False`).
+
+        Notes
+        -----
+    <<<<<<< Updated upstream
```
Comment (fschuch) [inline xcompact3d_toolbox/io.py:257]: Review docstring

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+visu = [
+    "matplotlib>=3.2",
+    "bokeh>=2.3",
+    "datashader>=0.13",
+    "hvplot>=0.7",
+    "panel>=0.12",
+    "holoviews>=1.14",
+]
+
+[project.urls]
```
Comment (fschuch) [inline pyproject.toml:45]: This is missing

File: docs/conf.py [hunk 3]
```diff
@@ -1,97 +0,0 @@
-# Configuration file for the Sphinx documentation builder.
```
Comment (fschuch) [inline docs/conf.py:1]: Need to review the config and docs in general the assert it works as expected after changing the framework. Specially the buttons to launch on binder.

File: docs/logo.png
Comment (fschuch) [inline docs/logo.png:1]: Remember to replace the logo

File: README.md [hunk 4]
```diff
@@ -78,95 +75,105 @@ Click on any link above to launch [Binder](https://mybinder.org/) and interact w
...
+- User interface for the parameters with IPywidgets:
 
-   ```python
-   prm = x3d.ParametersGui()
-   prm
-   ```
+  ```python
+  prm = x3d.ParametersGui()
+  prm
+  ```
```
[the remark this answers is not in the document]
Comment (fschuch) [reply README.md:172]: This was not a good suggestion. Big block where the change is not clear.

File: xcompact3d_toolbox/genepsi.py [hunk 5]
```diff
@@ -148,28 +148,18 @@ def fix(xi_in, xf_in, nobjx, nobjxraf, epsi, refepsi, nraf, nobjmax):
...
-                        and idebraf > ifinraf
-                        and isol == 1
-                    ):
+                    if idebraf != 0 and ifinraf != 0 and idebraf > ifinraf and isol == 1:
                         iobj += 1
                         for ii in range(iobj, nobjmax - 1):
                             xi[ii] = xi[ii + 1]
                         iobj -= 1
-                        for i in range(iobj, nobjmax - 1):
+                        for ii in range(iobj, nobjmax - 1):
```
[the remark this answers is not in the document]
Comment (fschuch) [reply xcompact3d_toolbox/genepsi.py:162]: This complicity is a known issue already tracked on sonarcloud, so I'm resolving the comment.

File: README.md [hunk 6]
```diff
@@ -1,9 +1,11 @@
...
+| Meta    | [![Wizard Template](https://img.shields.io/badge/Wizard-Template-%23447CAA)](https://github.com/fschuch/wizard-template) [![Checked with mypy](https://www.mypy-lang.org/static/mypy_badge.svg)](https://mypy-lang.org/) [![Hatch project](https://img.shields.io/badge/%F0%9F%A5%9A-Hatch-4051b5.svg)](https://github.com/pypa/hatch) [![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff) [![PyPI - License](https://img.shields.io/pypi/l/xcompact3d-toolbox?color=blue)](https://github.com/fschuch/xcompact3d_toolbox/blob/master/LICENSE) [![EffVer Versioning](https://img.shields.io/badge/version_scheme-EffVer-0097a7)](https://jacobtomlinson.dev/effver) |
```
[the remark this answers is not in the document]
Comment (fschuch) [reply README.md:8]: I agree, but this comes from pre-commit hooks, so let's keep as it is
